core/src/test/java/com/linecorp/armeria/client/HttpResponseWrapperTest.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.Test;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+import com.linecorp.armeria.client.HttpResponseDecoder.HttpResponseWrapper;
+import com.linecorp.armeria.common.CommonPools;
+import com.linecorp.armeria.common.HttpData;
+import com.linecorp.armeria.common.HttpHeaderNames;
+import com.linecorp.armeria.common.HttpHeaders;
+import com.linecorp.armeria.common.HttpMethod;
+import com.linecorp.armeria.common.HttpObject;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.stream.StreamMessage;
+import com.linecorp.armeria.internal.InboundTrafficController;
+
+import io.netty.channel.Channel;
+
+public class HttpResponseWrapperTest {
+
+    @Test
+    public void headersAndData() throws Exception {
+        final DecodedHttpResponse res = new DecodedHttpResponse(CommonPools.workerGroup().next());
+        final HttpResponseWrapper wrapper = httpResponseWrapper(res);
+
+        wrapper.tryWrite(HttpHeaders.of(200).addInt(HttpHeaderNames.CONTENT_LENGTH, "foo".length()));
+        wrapper.tryWrite(HttpData.ofUtf8("foo"));
+        wrapper.close();
+
+        final List<HttpObject> unaggregated = unaggregate(res);
+        assertThat(unaggregated).containsExactly(
+                HttpHeaders.of(200).addInt(HttpHeaderNames.CONTENT_LENGTH, 3),
+                HttpData.ofUtf8("foo"));
+    }
+
+    @Test
+    public void headersAndTrailers() throws Exception {
+        final DecodedHttpResponse res = new DecodedHttpResponse(CommonPools.workerGroup().next());
+        final HttpResponseWrapper wrapper = httpResponseWrapper(res);
+
+        wrapper.tryWrite(HttpHeaders.of(200));
+        wrapper.tryWrite(HttpHeaders.of(HttpHeaderNames.of("bar"), "baz"));
```

Comment:
Probably better making sure `tryWrite()` returns `true` in many places in this test case?